### ISSUE TITLE
WS07: Evaluation loop trains on the test set

### ISSUE THREAD

**Issue by SaifSyed08** [Graylab/DL4Proteins-notebooks#11]
The evaluation loop calls `loss.backward()` and `optimizer.step()` on the test set for 10 epochs.

`model.eval()` changes dropout and batch-normalization behavior, but it does not prevent parameter updates. This means the model is effectively trained on the test data before its accuracy is reported.

The evaluation should instead run once with:

```python
model.eval()

with torch.inference_mode():
    for data in test_loader:
        output = model(data)
        loss = criterion(output, data.graph_y)
```

The evaluation loop should not contain `backward()` or `optimizer.step()`.
